server/src/main/java/org/elasticsearch/cluster/routing/allocation/Explanations.java

```diff
@@ -0,0 +1,106 @@
+            Elasticsearch cannot allocate this shard because all the copies it found in the cluster are stale or corrupt. Elasticsearch \
+            will allocate this shard when a node holding a good copy of its data joins the cluster. If no such node is available then you \
+            should restore this index from a recent snapshot.""";
+
+        public static final String NO_COPIES = """
+            Elasticsearch cannot allocate this shard because it could not find any copies of its data in the cluster. Elasticsearch \
+            will allocate this shard when a node holding a good copy of its data joins the cluster. If no such node is available then you \
+            should restore this index from a recent snapshot.""";
+
+        public static final String DELAYED_WITH_ALTERNATIVE = """
+            The node holding this shard copy left the cluster and Elasticsearch is waiting for it to return. If the node does not return \
+            within [%s] then Elasticsearch will be able to allocate this shard to another node in the cluster.""";
+
+        public static final String DELAYED_WITHOUT_ALTERNATIVE = """
+            The node holding this shard copy left the cluster and Elasticsearch is waiting for it to return. If the node does not return \
+            within [%s] then Elasticsearch will not be able to allocate this shard to another node in the cluster.""";
+
+        public static final String EXISTING_STORES_FORBIDDEN = """
+            Elasticsearch is not permitted to allocate this shard to any of the nodes in the cluster that hold an in-sync copy of its \
+            data. If you expect this shard to be allocated to one of the nodes in the cluster then you should find this node in the \
+            node-by-node explanation and address the reasons which prevent Elasticsearch from allocating this shard there.""";
+
+        public static final String ALL_NODES_FORBIDDEN = """
+            Elasticsearch is not permitted to allocate this shard to any of the nodes in the cluster. If you expect this shard to be \
+            allocated to one of the nodes in the cluster then you should find this node in the node-by-node explanation and address the \
+            reasons which prevent Elasticsearch from allocating this shard there.""";
+
+    }
+
+    public static final class Rebalance {
+
+        public static final String YES = """
+            Elasticsearch can rebalance this shard onto a different node.""";
+
+        public static final String ALREADY_BALANCED = """
+            Elasticsearch cannot rebalance this shard onto a different node since there is no node to which allocation is permitted which \
+            would improve the cluster balance. If you expect this shard to be rebalanced onto one of the nodes in the cluster then you \
+            should find this node in the node-by-node explanation and address the reasons which prevent Elasticsearch from rebalancing \
+            this shard there.""";
```

Comment:
Again, the error message references a specific shard, but I don't think this message requires one.
```suggestion
            Elasticsearch can't rebalance this shard to another node. Although it would improve the cluster balance, Elasticsearch isn't \
            allowed to allocate the shard to any node in the cluster. If you expect the shard to be rebalanced to another node, review \
            the node-by-node explanation and address the reasons which prevent Elasticsearch from rebalancing the shard.""";
```

Reply:
Similarly, you need to choose a node to make progress.